src/Stack/Setup.hs

```diff
@@ -632,14 +635,15 @@ getGhcBuilds = do
         return builds
 
 #if !WINDOWS
--- | Encode an OpenBSD version (like "6.1") into a valid argument for
+-- | Encode an OpenBSD/FreeBSD version (like "6.1"/"12.0-RELEASE") into a valid argument for
 -- CompilerBuildSpecialized, so "maj6-min1". Later version numbers are prefixed
 -- with "r".
 -- The result r must be such that "ghc-" ++ r is a valid package name,
 -- as recognized by parsePackageNameFromString.
 mungeRelease :: String -> String
-mungeRelease = intercalate "-" . prefixMaj . splitOn "."
+mungeRelease = intercalate "-" . prefixMaj . cutOffFreeBSDSuffix . splitOn "."
   where
+    cutOffFreeBSDSuffix = map $ head . splitOn "-RELEASE" . head . splitOn "-CURRENT" . head . splitOn "-STABLE"
```

Comment:
Can you talk through what you intend this code to do?

Reply:
This cuts off "-RELEASE", "-CURRENT" and "-STABLE" suffixes, that are present in output of `sysRelease` on FreeBSD.

Reply:
Would it be safe to just cut off the `-[A-Z]+$` regex rather than hardcode the specific `-RELEASE`, `-CURRENT`, and `-STABLE` suffixes?

Reply:
Now you mention that, right. There are `-ALPHA1`, `BETA2` and other suffixes, so it is better to just cut off using regex. What package/function should I use for that?

Reply:
It's probably easier to use standard text functions like `dropWhileEnd` rather than resorting to regexes, but we are using the `regex-applicative-text` in other parts of the codebase.